src/integrationTests/async.spec.ts

```diff
@@ -175,3 +195,165 @@ describe('when window.__uid2.init is called on SdkLoaded from a callback', () =>
     expect(sdkWindow.__uid2.getAdvertisingToken()).toBe(identity.advertising_token);
   });
 });
+
+describe('when getFreshAdvertisingToken is called before init complete', () => {
+  describe('when initialising with a valid identity', () => {
+    const identity = makeIdentity();
+    test('it should resolve promise after invoking the callback', () => {
```

Comment:
Why does it matter that the getFreshAdvertisingToken promise is resolved after calling the init callback?